src/nuclide_header.F90

```diff
@@ -265,11 +267,31 @@ subroutine nuclide_from_hdf5(this, group_id, temperature, method, tolerance)
                &for " // trim(this % name) // " at or near " // &
                trim(to_str(nint(temp_desired))) // " K.")
         end if
-      end do TEMP_LOOP
+      end do
 
     case (TEMPERATURE_INTERPOLATION)
-      ! TODO: Get bounding temperatures
-      call fatal_error("Temperature interpolation not yet implemented")
+      ! If temperature interpolation or multipole is selected, get a list of
+      ! bounding temperatures for each actual temperature present in the model
+      TEMP_LOOP: do i = 1, temperature % size()
+        temp_desired = temperature % data(i)
+
+        do j = 1, size(temps_available) - 1
```

Comment:
Right now I would expect it to fail, because for each actual temperature in your model it searches for two bounding temperatures. Not sure how much I'd have to hack things to get it to work with a library that has a single temperature, but it's worth a thought.